### PR TITLE
Enhance picking valid xlMeta based on quorum

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Enhance picking valid xlMeta based on quorum

## Motivation and Context
This PR borrows the idea from getFormatXLQuorum()

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/xl-v1-multipart.go [hunk 1]
```diff
@@ -702,22 +702,26 @@ func (xl xlObjects) CompleteMultipartUpload(ctx context.Context, bucket string,
...
+		finalParts[i] = objectPartInfo{
 			Number: part.PartNumber,
 			ETag:   part.ETag,
 			Size:   currentXLMeta.Parts[partIdx].Size,
 			Name:   fmt.Sprintf("part.%d", part.PartNumber),
 		}
 	}
 
+	// Save final parts list.
+	xlMeta.Parts = make([]objectPartInfo, len(parts))
```
Comment (donatello) [inline cmd/xl-v1-multipart.go]: This allocation seems unnecessary - it is alreadying happening in line 713 above.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by fwessels — APPROVED** [review]
LGTM

**Review by donatello — APPROVED** [review]
It looks correct but appears unnecessary - quorum of mod times should be enough. Additionally part names are unique already - e.g. part.3 for part 3 so the hashing doesn't seem to be needed. We can relook at improving this IMHO.

**Pull request merged**